cmake/clang-tidy.cmake

```diff
@@ -0,0 +1,36 @@
+option(ENABLE_CLANG_TIDY "Add run-clang-tidy automatically to builds" OFF)
+find_program(CLANG_TIDY_EXE
```

Comment:
this blck untill the end should be executed only if tidy option in ON

Reply:
wait a moment ...

There are tree way to work:

as pre compile step:
```shell
[1/26] /opt/local/bin/cmake -E __run_co_compile --launcher=ccache --tidy=/opt/local/bin/clang-tidy --source=/Users/clausklein/Workspace/cpp/spdlog/src/spdlog.cpp -- /opt/local/bin/g++  -DSPDLOG_COMPILED_LIB -I/Users/clausklein/Workspace/cpp/spdlog/include -g   -Wall -Wextra -Wconversion -pedantic -Wfatal-errors -std=c++11 -MD -MT CMakeFiles/spdlog.dir/src/spdlog.cpp.o -MF CMakeFiles/spdlog.dir/src/spdlog.cpp.o.d -o CMakeFiles/spdlog.dir/src/spdlog.cpp.o -c /Users/clausklein/Workspace/cpp/spdlog/src/spdlog.cpp
[2/26] /opt/local/bin/cmake -E __run_co_compile --launcher=ccache --tidy=/opt/local/bin/clang-tidy --source=/Users/clausklein/Workspace/cpp/spdlog/src/stdout_sinks.cpp -- /opt/local/bin/g++  -DSPDLOG_COMPILED_LIB -I/Users/clausklein/Workspace/cpp/spdlog/include -g   -Wall -Wextra -Wconversion -pedantic -Wfatal-errors -std=c++11 -MD -MT CMakeFiles/spdlog.dir/src/stdout_sinks.cpp.o -MF CMakeFiles/spdlog.dir/src/stdout_sinks.cpp.o.d -o CMakeFiles/spdlog.dir/src/stdout_sinks.cpp.o -c /Users/clausklein/Workspace/cpp/spdlog/src/stdout_sinks.cpp
```
as target check:

```shell
cmake --build ../.build-spdlog-Debug --target check | tee run-clang-tidy.log 2>&1
```

and on console with run-clang-tidy.py which needs the compile_commands.json too
```shell
run-clang-tidy.py example/*.cpp
```